### PR TITLE
fix: array coverity hits

### PR DESCRIPTION
**Pull request by ethanglaser**
## Description

Addresses 2 array hits in coverity.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

### CODE REVIEW

**Comment by ethanglaser** [comment]
CI fails unrelated, to be fixed in https://github.com/intel/scikit-learn-intelex/pull/2190

**Comment by david-cortes-intel** [comment]
Error from the public CI job reads as follows:
```
Info from 'helper_mpi_tests.py'
sys.argv: ['/home/vsts/work/1/s/sklearnex/tests/helper_mpi_tests.py', 'pytest', '-k', 'spmd', '--with-mpi', '--verbose', '--pyargs', 'sklearnex']
ERROR: usage: helper_mpi_tests.py [options] [file_or_dir] [file_or_dir] [...]
helper_mpi_tests.py: error: unrecognized arguments: --with-mpi
  inifile: /home/vsts/work/1/s/sklearnex/pyproject.toml
  rootdir: /home/vsts/work/1/s/sklearnex
```

It means the job doesn't have `pytest-mpi` installed, which is required for running the SPMD tests.

**Review by Vika-F — APPROVED** [review]
LGTM

File: cpp/oneapi/dal/array.hpp [hunk 1]
```diff
@@ -222,6 +222,11 @@ class array {
         other.reset_data();
     }
 
+    // Add copy assignment, move assignment, and destructor
+    array<T>& operator=(const array<T>& other) = default;
+    array<T>& operator=(array<T>&& other) = default;
```
Comment (ethanglaser) [inline cpp/oneapi/dal/array.hpp:227]:
```suggestion
    // Add destructor
```

File: cpp/oneapi/dal/chunked_array.hpp [hunk 2]
```diff
@@ -50,6 +50,23 @@ class ONEDAL_EXPORT chunked_array : public detail::chunked_array_base {
...
+        base_t::operator=(std::forward<base_t>(array));
+        return *this;
+    }
+
+    /// @brief       Copy assignment
+    /// @param array Source array
+    chunked_array& operator=(const chunked_array& array) {
+        base_t::operator=(array);
+        return *this;
+    }
```
Comment (Vika-F) [inline cpp/oneapi/dal/chunked_array.hpp:65]: Should be equivalent but with less lines of code.

```suggestion
    chunked_array& operator=(chunked_array&& array) = default;

    /// @brief       Copy assignment
    /// @param array Source array
    chunked_array& operator=(const chunked_array& array) = default;
```

**Pull request merged**